ui/pages/confirmations/components/confirm/info/approve/approve.tsx

```diff
@@ -75,12 +104,110 @@ const ApproveStaticSimulation = () => {
       title={t('simulationDetailsTitle')}
       titleTooltip={t('simulationDetailsTitleTooltip')}
       description={t('simulationDetailsApproveDesc')}
-      simulationHeading={t('simulationApproveHeading')}
+      simulationHeading={
+        isNFT ? t('simulationApproveHeading') : t('spendingCap')
+      }
       simulationElements={simulationElements}
     />
   );
 };
 
+const SpendingCapGroup = ({
+  tokenSymbol,
+  decimals,
+}: {
+  tokenSymbol: string;
+  decimals: string;
+}) => {
+  const t = useI18nContext();
+
+  const transactionMeta = useSelector(
+    currentConfirmationSelector,
+  ) as TransactionMeta;
+
+  const { tokenAmount, formattedTokenNum, value } = useApproveTokenSimulation(
+    transactionMeta,
+    decimals,
+  );
+
+  const SpendingCapElement = (
+    <ConfirmInfoRowText
+      text={
+        tokenAmount === UNLIMITED_MSG
+          ? `${t('unlimited')} ${tokenSymbol}`
+          : `${formattedTokenNum} ${tokenSymbol}`
+      }
+      onEditClick={
+        transactionMeta.type === TransactionType.tokenMethodIncreaseAllowance
+          ? () => console.log('TODO on a following ticket')
+          : undefined
+      }
+      editIconClassName="edit-spending-cap"
+    />
+  );
```

Comment:
Putting JSX in variable is INHO not great idea, any reason for not including it inline.

Reply:
Yes, the idea is to reduce code duplication in the following code block

```jsx
      <ConfirmInfoRow
        label={t('spendingCap')}
        tooltip={t('spendingCapTooltipDesc')}
      >
        {tokenAmount === UNLIMITED_MSG ? (
          <Tooltip title={formattedTokenNum}>{SpendingCapElement}</Tooltip>
        ) : (
          SpendingCapElement
        )}
      </ConfirmInfoRow>
```

This was changed to address feedback on this comment https://github.com/MetaMask/metamask-extension/pull/26606#discussion_r1727353042